### ISSUE TITLE
[Bug]: run vllm api with docker

### ISSUE THREAD

**Issue by shudct** [vllm-project/vllm#4940]
### Your current environment

docker images :vllm/vllm-openai:latest

### 🐛 Describe the bug

`docker run --runtime nvidia --gpus all -v ~/.cache/huggingface:/root/.cache/huggingface -v /mnt/ddata/models/Baichuan2-13B-Chat:/Baichuan2-13B-Chat --env "HUGGING_FACE_HUB_TOKEN=<secret>" -p 5000:5000 --ipc=host vllm/vllm-openai:latest --model /Baichuan2-13B-Chat --trust-remote-code --tensor-parallel-size 2`

I use the command above to test and occur "RuntimeError: Unexpected error from cudaGetDeviceCount(). Did you run some cuda functions before calling NumCudaDevices() that might have already set an error? Error 804: forward compatibility was attempted on non supported HW"

how to solve this problem, thx.

**Comment by guliujian** [comment]
what is your graphic card and the host driver version and the host cuda version?

**Comment by josephrocca** [comment]
Yep, this issue seems to be due to an old host driver version. I get the error for machines where `nvidia-smi` says driver version 525, but not when it says 550. But note that this is likely to only be a problem for non-datacenter-grade GPUs:

https://github.com/pytorch/pytorch/issues/40671#issuecomment-904794431

> When the host outside the docker is running a driver that doesn't support the cuda version inside the container, you can see this message if you don't have a datacenter/Tesla GPU.

And unfortunately, as of writing, there are a bunch of machines on Runpod (for example) which have the 525 driver version, so when testing the latest image (`vllm/vllm-openai:v0.4.3`) on Runpod 4090s, it will randomly work or not work depending on what machine you're assigned in deployment.

I'm guessing this isn't fixable by vllm, since it unfortunately seems like something that must be "solidified" at docker image compile time? Or is there a way to fallback at runtime in order to support older host driver versions?

**Comment by taco-devs** [comment]
Looks like this is solved on #5510 

but my A100 SXM is updated to the 550 driver version and i can't run this. Any idea on how to fix this ?

<img width="866" alt="image" src="https://github.com/user-attachments/assets/a84814e1-8a44-47ae-bed5-d6d6e3416624">
